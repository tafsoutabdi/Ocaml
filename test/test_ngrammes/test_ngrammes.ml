open Tokenization.Ngrammes
open Utils.Std

let runtest k l to_string =
  Format.printf "l = %s@.ngrammes %d l = %s@.@."
    (string_of_list to_string l)
    k 
    (
      try string_of_list (string_of_list to_string) (ngrammes k l)
      with Invalid_argument s -> Format.sprintf "Invalid_argument (\"%s\")" s
    )

let () = runtest 3 ['a'; 'b'; 'c'; 'd'; 'e'; 'f'] string_of_char

let () = runtest 2 ["longtemps"; "je"; "me"; "suis"; "couchÃ©"] string_of_string

let () = runtest 10 ['a'; 'b'; 'c'; 'd'; 'e'; 'f'] string_of_char

let () = runtest 1 ['a'; 'b'; 'c'; 'd'; 'e'; 'f'] string_of_char

let () = runtest 1 [] string_of_char

let () = runtest (-1) ['a'; 'b'; 'c'; 'd'; 'e'; 'f'] string_of_char

let () = runtest 0 ['a'; 'b'; 'c'; 'd'; 'e'; 'f'] string_of_char
