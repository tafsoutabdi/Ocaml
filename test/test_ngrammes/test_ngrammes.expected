l = ['a'; 'b'; 'c'; 'd'; 'e'; 'f']
ngrammes 3 l = [['a'; 'b'; 'c']; ['b'; 'c'; 'd']; ['c'; 'd'; 'e']; ['d'; 'e'; 'f']]

l = ["longtemps"; "je"; "me"; "suis"; "couché"]
ngrammes 2 l = [["longtemps"; "je"]; ["je"; "me"]; ["me"; "suis"]; ["suis"; "couché"]]

l = ['a'; 'b'; 'c'; 'd'; 'e'; 'f']
ngrammes 10 l = []

l = ['a'; 'b'; 'c'; 'd'; 'e'; 'f']
ngrammes 1 l = [['a']; ['b']; ['c']; ['d']; ['e']; ['f']]

l = []
ngrammes 1 l = []

l = ['a'; 'b'; 'c'; 'd'; 'e'; 'f']
ngrammes -1 l = Invalid_argument ("ngrammes")

l = ['a'; 'b'; 'c'; 'd'; 'e'; 'f']
ngrammes 0 l = Invalid_argument ("ngrammes")

